docs/blog/posts/2024/07-26 asynchronous message support.md

```diff
@@ -0,0 +1,144 @@
+    process_message(message)
+
+pact = Pact(
+    consumer="deleteUserService",
+    provider="someProvider",
+).with_specification("V3")  # (1)
+
+(
+    pact
+    .upon_receiving("a request to delete a user", "Async")
+    .with_body(
+        json.dumps({
+            "action": "delete_user",
+            "user_id": "123",
+        })
+    )  # (2)
+)
+
+pact.verify(handler, "Async")
+```
+
+1.  Support for asynchronous messages starts in version 3 of the Pact specification.
+2.  No `will_respond_with` method exists for asynchronous messages since there’s no response expected.
+
+This example highlights how the verification of asynchronous messages differs from HTTP interaction. As the transport layer is abstracted away, a `handler` function is required to parse the raw message string or bytes, and pass it to the underlying function that processes the message.
+
+The `handler` would also typically be responsible for mocking the underlying systems that the consumer interacts with, such as the database or file system. This allows the consumer to be tested in isolation, without relying on external services. Furthermore, the mocked systems can then be inspected to verify that the consumer has performed the expected actions.
+
+## Provider Example
+
+For context of asynchronous messages, the provider is the service that sends the message and might be referred to as the _publisher_ or _producer_. Since the contract is defined by the consumer, the Pact provider test simply has to verify that the messages it sends meet the expectations of the consumer.
+
+As the underlying protocol is abstracted away, Pact uses a local HTTP server to receive the messages that the provider sends. The provider test for the above consumer might look something like this:
+
+```python
+from pact.v3 import Verifier
+
+class Provider:
+    """
+    A simple HTTP provider that sends messages to the consumer.
+
+    This would typically use the same underlying functions that would generate messages, except that instead of being sent into the message queue, they are sent to the consumer's HTTP server.
+    """
+
+provider = Provider()
+
+(
+    Verifier()
+    .set_info("someProvider", url=provider.url)  # (1)
+    .set_source("/path/to/pacts")
+    .set_state(provider.state_url)  # (2)
+    .add_transport(  # (3)
+        protocol="message",
+        path="/_pact/message",
+    )
+ )
+```
+
+1.  The provider URL is required but only used if generating both HTTP and message interactions.
+2.  The provider URL is required, though would only be used if the provider generates both HTTP and message interactions.
+3.  This path is used by Pact to ensure that the provider is in the correct state before sending the message.
+
+Those familiar with HTTP interactions will notice that the process is very similar, with the key difference of the additional `add_transport` method. This configure a HTTP endpoint on the provider that Pact will call to trigger the provider to send the message.
```

Comment:
configures

Reply:
>1.  The provider URL is required but only used if generating both HTTP and message interactions.
> 2.  The provider URL is required, though would only be used if the provider generates both HTTP and message interactions.

are the provider.url and provider.state_url only used for http, will they be ignored for message interactions?

mentioned in points 1/2 above.

for 3, `path="/_pact/message"`

> This configure a HTTP endpoint on the provider

If we are validating a message interaction only, no http interactions. I assume we need 1. the provider url to prefix this path

Reply:
For your first part: I've disambiguated the state_url portion to make it clear it is for _both_.

For your second part: You have to define a `messages` transport at home point, though it technically could be done at the first. I'll rework the example to highlight this.